with-foundry/README.md

```diff
@@ -24,12 +24,27 @@ Install [noirup](https://noir-lang.org/docs/getting_started/installation/#instal
    noirup
    ```
 
-3. Install foundryup and follow the instructions on screen. You should then have all the foundry tools like `forge`, `cast`, `anvil` and `chisel`.
+3. Install foundryup and follow the instructions on screen. You should then have all the foundry
+   tools like `forge`, `cast`, `anvil` and `chisel`.
 
 ```bash
 curl -L https://foundry.paradigm.xyz | bash
 ```
 
+4. Install the correct version of the
+   [Barretenberg](https://github.com/AztecProtocol/aztec-packages/tree/master/barretenberg/cpp/src/barretenberg/bb#version-compatibility-with-noir)
+   proving backend for Noir (bb).
+
+   ```bash
+   curl -L https://raw.githubusercontent.com/AztecProtocol/aztec-packages/master/barretenberg/cpp/installation/install | bash
+   ```
+
+   then
+
+   ```bash
+   bbup -v 0.46.1 # compatible with nargo 0.31.0
```

Comment:
```suggestion
   bbup -v 0.41.0 # compatible with nargo 0.31.0
```

For Noir 0.31.0, the compatible bb version is 0.41.0

Reply:
good catch